rust/datafusion/src/physical_plan/common.rs

```diff
@@ -31,53 +32,58 @@ use array::{
 };
 use arrow::datatypes::{DataType, SchemaRef};
 use arrow::error::Result as ArrowResult;
-use arrow::record_batch::{RecordBatch, RecordBatchReader};
+use arrow::record_batch::RecordBatch;
 use arrow::{
     array::{self, ArrayRef},
     datatypes::Schema,
 };
+use futures::{Stream, TryStreamExt};
 
-/// Iterator over a vector of record batches
-pub struct RecordBatchIterator {
+/// Stream of record batches
+pub struct SizedRecordBatchStream {
```

Comment:
I was thinking to reserve `Buffered` to a stream that buffers N entries. This one is already initialized with everything via `new`. This is more like the [`iter`](https://docs.rs/futures/0.3.6/futures/stream/fn.iter.html), with a fixed known len and a known schema. IMO we should avoid using this struct, as it requires us to collect to it somewhere.

Reply:
Yeah, I was thinking it would be nice to imply in this struct's name somehow that it should be avoided if possible, because, as you say, it requires buffering / collecting up the `RecordBatch`es before producing any results